### PR TITLE
adapt kubeclient changes

### PR DESCRIPTION
**Pull request by moolitayer**
adapt kubeclient changes from kubeclient pr#92[1]

adapt kubeclient changes from kubeclient pr#114[2]

[1] abonas/kubeclient#92
[2] abonas/kubeclient#114

### CODE REVIEW

**Comment by moolitayer** [comment]
marking as wip until kubeclient version is known

**Comment by moolitayer** [comment]
fyi @abonas @simon3z

File: lib/openshift_client.rb [hunk 1]
```diff
@@ -20,17 +20,50 @@ class Client < Kubeclient::Common::Client
...
+      bearer_token(auth_options[:bearer_token]) if auth_options[:bearer_token]
     end
 
     def all_entities
       retrieve_all_entities(ENTITY_TYPES)
     end
 
+    private
+
+    def validate_auth_options(opts)
```
Comment (simon3z) [inline lib/openshift_client.rb]: @moolitayer any way to reuse the kubeclient `validate_auth_options`? (instead of copy/paste)

Comment (moolitayer) [inline lib/openshift_client.rb]: yes and we now need:
https://github.com/abonas/kubeclient/pull/109

**Comment by moolitayer** [comment]
this now depends on abonas/kubeclient#109 too

**Comment by moolitayer** [comment]
@aboans @simon3z  please review.

File: openshift_client.gemspec [hunk 2]
```diff
@@ -24,7 +24,7 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency 'minitest'
   spec.add_development_dependency 'webmock'
   spec.add_development_dependency 'rubocop', '= 0.30.0'
-  spec.add_dependency 'kubeclient', '>= 0.1.17'
+  spec.add_dependency 'kubeclient', '>= 0.3.1'
```
Comment (abonas) [inline openshift_client.gemspec]: relevant gem ver is now 0.4.0 (released)

Comment (moolitayer) [inline openshift_client.gemspec]: updated

**Comment by moolitayer** [comment]
removing wip since kubeclient 0.4.0 has been released

**Comment by moolitayer** [comment]
changed

```
-                   auth_options: {
-                     user: nil,
-                     password: nil,
-                     bearer_token: nil
-                   })
+                  auth_options: {})
```

to comply with verify_ssl

**Comment by moolitayer** [comment]
fyi @abonas

File: lib/openshift_client.rb [hunk 3]
```diff
@@ -20,11 +20,28 @@ class Client < Kubeclient::Common::Client
...
+
+      validate_auth_options(auth_options)
+
       handle_uri(uri, path)
       @api_version = version
       @headers = {}
-      ssl_options
+      @ssl_options = ssl_options
+
+      @basic_auth_user = auth_options[:user]
```
Comment (abonas) [inline lib/openshift_client.rb]: the code in lines 42-44 differs from similar code in kubeclient and provides a slightly different handling logic (and doesn't handle bearer token file), is this on purpose?
cc @oschreib @jimmidyson

Comment (moolitayer) [inline lib/openshift_client.rb]: Right. 
This patch includes the mandatory changes openshift_client needed to not break 
using kubeclient after abonas/kubeclient#92
It does not included support for bearer token file added by @jimmidyson in https://github.com/abonas/kubeclient/pull/98

Comment (abonas) [inline lib/openshift_client.rb]: > Right.
> This patch includes the mandatory changes openshift_client needed to not break
> using kubeclient after abonas/kubeclient#92
> It does not included support for bearer token file added by @jimmidyson in abonas/kubeclient#98

but even without bearer token file, the basic auth handling code is different - here it is always assigned.

Comment (moolitayer) [inline lib/openshift_client.rb]: Pretty sure those changes where made in the same pr.

let me know if you need anything from me to move this patch.

Comment (abonas) [inline lib/openshift_client.rb]: > Pretty sure those changes where made in the same pr.
> 
> let me know if you need anything from me to move this patch.

please look here
https://github.com/abonas/kubeclient/blob/master/lib/kubeclient.rb#L51
as you can see, there's a conditional there that checks if basic auth details should be assigned or not.
in your current patch here, you are always assigning basic auth details, while the bearer token is assigned conditionally - hence the logic is different.

Comment (moolitayer) [inline lib/openshift_client.rb]: On Tue, Aug 25, 2015 at 2:50 PM, abonas notifications@github.com wrote:

> In lib/openshift_client.rb
> https://github.com/abonas/openshift_client/pull/10#discussion_r37856078:
> 
> > ```
> >    handle_uri(uri, path)
> >    @api_version = version
> >    @headers = {}
> > ```
> > -      ssl_options
> > -      @ssl_options = ssl_options
> >   +
> > -      @basic_auth_user = auth_options[:user]
> 
> Pretty sure those changes where made in the same pr.
> 
> let me know if you need anything from me to move this patch.
> 
> please look here
> https://github.com/abonas/kubeclient/blob/master/lib/kubeclient.rb#L51
> as you can see, there's a conditional there that checks if basic auth
> details should be assigned or not.
> in your current patch here, you are always assigning basic auth details,
> while the bearer token is assigned conditionally - hence the logic is
> different.

That change in logic was done as part of dyson bearer patch
(not saying it's related to the bearer patch but it was 'snuck in' together
)

so it is an additional change.

**Comment by abonas** [comment]
@simon3z please also review. 
I've expressed my concern regarding the addition of different logic for basic auth handling 
vs the one present in kubeclient above.

**Comment by simon3z** [comment]
@abonas @moolitayer let's move:

https://github.com/abonas/kubeclient/blob/master/lib/kubeclient.rb#L51-L59

to a common place (`common.rb` in kubeclient) and let's reuse it from here.

**Comment by abonas** [comment]
> @abonas @moolitayer let's move:
> 
> https://github.com/abonas/kubeclient/blob/master/lib/kubeclient.rb#L51-L59
> 
> to a common place (common.rb in kubeclient) and let's reuse it from here.

:+1:

**Comment by moolitayer** [comment]
marking as wip until https://github.com/abonas/kubeclient/pull/114 matures

**Comment by moolitayer** [comment]
> marking as wip until abonas/kubeclient#114 matures

removing wip, kubeclient 0.5.0 with abonas/kubeclient#114  has been released.

**Comment by simon3z** [comment]
The auth logic is now completely shared with kubeclient. Merging as master is currently broken without this PR.

**Pull request merged**